src/main/java/com/iota/iri/service/ledger/impl/LedgerServiceImpl.java

```diff
@@ -173,11 +184,16 @@ public boolean isBalanceDiffConsistent(Set<Hash> approvedHashes, Map<Hash, Long>
                             if (transactionViewModel.getCurrentIndex() == 0) {
                                 boolean validBundle = false;
 
-                                final List<List<TransactionViewModel>> bundleTransactions = BundleValidator.validate(
+                                final List<List<TransactionViewModel>> bundleTransactions = bundleValidator.validate(
                                         tangle, snapshotProvider.getInitialSnapshot(), transactionViewModel.getHash());
 
                                 for (final List<TransactionViewModel> bundleTransactionViewModels : bundleTransactions) {
 
+                                    //ISSUE 1008: generateBalanceDiff should be refactored so we don't have those hidden
+                                    // concerns
+                                    spentAddressesService
```

Comment:
small caveat, you will also scan addresses that are not spent for sure.
for example data transactions. if probably worth first filtering only negative values and if none empty submit.

Reply:
Yup,
We are currently submitting empty sets after filtering. There is no logical fallacy but it does result in creating more objects thus increasing GC pressure...

I would say that this is a point we need to pay attention to in general with the Data Providers